src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageSolutionDetailControlModel.cs

```diff
@@ -143,28 +143,21 @@ private async Task UpdateInstalledVersionsAsync(CancellationToken cancellationTo
             {
                 try
                 {
-                    (IPackageReferenceContextInfo TopLevelPackage, IPackageReferenceContextInfo TransitivePackage) installedPackage = await GetInstalledAndTransitivePackagesAsync(project.NuGetProject, Id, cancellationToken);
-                    var topLevelPackageVersion = installedPackage.TopLevelPackage;
-                    var transitivePackageVersion = installedPackage.TransitivePackage;
+                    IPackageReferenceContextInfo packageContext = await GetPackageContextAsync(project.NuGetProject, Id, cancellationToken);
 
                     project.InstalledVersionMaxVulnerability = -1;
                     project.RequestedVersion = null;
                     project.AutoReferenced = false;
 
-                    if (transitivePackageVersion != null)
+                    if (packageContext == null)
                     {
-                        project.PackageLevel = PackageLevel.Transitive;
-                        UpdateProjectInstallationInfo(project, transitivePackageVersion, installedVersionsSet);
-                    }
-                    else if (topLevelPackageVersion != null)
-                    {
-                        project.PackageLevel = PackageLevel.TopLevel;
-                        UpdateProjectInstallationInfo(project, topLevelPackageVersion, installedVersionsSet);
+                        project.InstalledVersion = null;
+                        project.PackageLevel = null;
                     }
                     else
                     {
-                        project.InstalledVersion = null;
-                        project.PackageLevel = null;
+                        project.PackageLevel = packageContext is ITransitivePackageReferenceContextInfo ? PackageLevel.Transitive : PackageLevel.TopLevel;
```

Comment:
Why set PackageLevel outside of the helper method `UpdateProjectInstallationInfo`?